### PR TITLE
Update s2i tasks documentation

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

- Small fix on the parameter name (and comments), `BUILDER_IMAGE` and
  `S2I_IMAGE` were inverted.
- Remove the `examples` folder and add an example inline.
- Add a non-exhaustive list of s2i builder images that can be used
  with this task.

cc @siamaksade @khrm 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: s2i/README.md [hunk 1]
```diff
@@ -13,15 +13,18 @@ kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/s2i/s
 
 ### Parameters
 
-* **BUILDER_IMAGE**: The name of the image containing the s2i command. See
+* **BUILDER_IMAGE**: The location of the s2i builder image.
```
Comment (siamaksade) [inline s2i/README.md]: Don't think this param is needed. It has to always be the same image

Comment (vdemeester) [reply s2i/README.md]: you mean the `S2I_IMAGE` image right (the one that has the `s2i` binary in) ?

File: s2i/README.md [hunk 2]
```diff
@@ -1,27 +1,33 @@
-# S2I
+# Source-to-Image
 
-This Task builds source into a container image using Project Atomic's
-[source-to-image/s2i](https://github.com/openshift/source-to-image). 
+[Source-to-Image (S2I)](https://github.com/openshift/source-to-image)
+is a toolkit and workflow for building reproducible container images
+from source code. S2I produces images by injecting source code into a
```
Comment (chmouel) [inline s2i/README.md]: nice one!

File: s2i/s2i.yaml [hunk 3]
```diff
@@ -28,9 +24,9 @@ spec:
 
   steps:
   - name: generate
-    image: ${inputs.params.BUILDER_IMAGE}
-    workingdir: /workspace/git-source
-    command: ['s2i', 'build', '${inputs.params.PATH_CONTEXT}', '${inputs.params.S2I_IMAGE}', '--as-dockerfile', '/gen-source/Dockerfile.gen']
+    image: quay.io/openshift-pipeline/s2i
```
Comment (chmouel) [inline s2i/s2i.yaml]: So we don't want to allow the user to override the s2i image?

Comment (vdemeester) [reply s2i/s2i.yaml]: Do you think we should do ? looking around catalog, sometimes we allow it, sometimes we don't, so I wasn't sure :joy_cat:

Comment (chmouel) [reply s2i/s2i.yaml]: If I was an advanced user using this I would probably would override it with my own image that would do fancy stuff, but i guess we are not to that point yet,

**Pull request merged**